jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererGoCodeScanTest.kt

```diff
@@ -0,0 +1,200 @@
+            filesInZip += 1
+        }
+
+        assertThat(filesInZip).isEqualTo(3)
+    }
+
+    @Test
+    fun `test getSourceFilesUnderProjectRoot`() {
+        assertThat(sessionConfigSpy.getSourceFilesUnderProjectRoot(mainGo).size).isEqualTo(3)
+    }
+
+    @Test
+    fun `test parseImport()`() {
+        val mainGoImports = sessionConfigSpy.parseImports(mainGo)
+        assertThat(mainGoImports.size).isEqualTo(2)
+
+        val helpGoImports = sessionConfigSpy.parseImports(helpGo)
+        assertThat(helpGoImports.size).isEqualTo(1)
+
+        val numberGoImports = sessionConfigSpy.parseImports(numberGo)
+        assertThat(numberGoImports.size).isEqualTo(1)
+    }
+
+    @Test
+    fun `test getImportedFiles()`() {
+        val files = sessionConfigSpy.getImportedFiles(mainGo, setOf())
+        assertNotNull(files)
+        assertThat(files).hasSize(2)
+        assertThat(files).contains(helpGo.path)
+        assertThat(files).contains(numberGo.path)
+    }
+
+    @Test
+    fun `test includeDependencies()`() {
+        val payloadMetadata = sessionConfigSpy.includeDependencies()
+        assertNotNull(payloadMetadata)
+        assertThat(sessionConfigSpy.isProjectTruncated()).isFalse
+        assertThat(payloadMetadata.sourceFiles.size).isEqualTo(3)
+        assertThat(payloadMetadata.payloadSize).isEqualTo(totalSize)
+        assertThat(payloadMetadata.linesScanned).isEqualTo(totalLines)
+        assertThat(payloadMetadata.buildPaths).hasSize(0)
+    }
+
+    @Test
+    fun `selected file larger than payload limit throws exception`() {
+        sessionConfigSpy.stub {
+            onGeneric { getPayloadLimitInBytes() }.thenReturn(100)
+        }
+        assertThrows<CodeWhispererCodeScanException> {
+            sessionConfigSpy.createPayload()
+        }
+    }
+
+    @Test
+    fun `test createPayload with custom payload limit`() {
+        sessionConfigSpy.stub {
+            onGeneric { getPayloadLimitInBytes() }.thenReturn(300)
+        }
+        val payload = sessionConfigSpy.createPayload()
+        assertNotNull(payload)
+        assertThat(sessionConfigSpy.isProjectTruncated()).isTrue
+        assertThat(payload.context.totalFiles).isEqualTo(2)
+
+        assertThat(payload.context.scannedFiles.size).isEqualTo(2)
+        assertThat(payload.context.scannedFiles).containsExactly(mainGo, helpGo)
+
+        assertThat(payload.context.srcPayloadSize).isEqualTo(220)
+        assertThat(payload.context.language).isEqualTo(CodewhispererLanguage.Go)
+        assertThat(payload.context.totalLines).isEqualTo(17)
+        assertNotNull(payload.srcZip)
+
+        val bufferedInputStream = BufferedInputStream(payload.srcZip.inputStream())
+        val zis = ZipInputStream(bufferedInputStream)
+        var filesInZip = 0
+        while (zis.nextEntry != null) {
+            filesInZip += 1
+        }
+
+        assertThat(filesInZip).isEqualTo(2)
+    }
+
+    @Test
+    fun `e2e happy path integration test`() {
```

Comment:
What does this test to? If it's an integration test, this should probably not be here?

Reply:
Looks like it's testing the `.run()` function with this SessionConfig class. It's not a true "integration test" but it does more than just testing each component in isolation